src/amd/compiler/aco_opcodes.py

```diff
@@ -1269,30 +1291,30 @@ def opcode(name, opcode_gfx9 = -1, format = Format.PSEUDO, input_mod = False, ou
 }
 # (gfx6, gfx7, gfx8, gfx9, gfx10, name) = (code, code, code, code, code, name)
 for (code, name) in IMAGE:
-   opcode(name, code, Format.MIMG)
+   opcode(name, code, code, Format.MIMG)
 
 IMAGE_ATOMIC = {
-   (0x0f, 0x0f, 0x10, "image_atomic_swap"),
-   (0x10, 0x10, 0x11, "image_atomic_cmpswap"),
-   (0x11, 0x11, 0x12, "image_atomic_add"),
-   (0x12, 0x12, 0x13, "image_atomic_sub"),
-   (0x13,   -1,   -1, "image_atomic_rsub"),
-   (0x14, 0x14, 0x14, "image_atomic_smin"),
-   (0x15, 0x15, 0x15, "image_atomic_umin"),
-   (0x16, 0x16, 0x16, "image_atomic_smax"),
-   (0x17, 0x17, 0x17, "image_atomic_umax"),
-   (0x18, 0x18, 0x18, "image_atomic_and"),
-   (0x19, 0x19, 0x19, "image_atomic_or"),
-   (0x1a, 0x1a, 0x1a, "image_atomic_xor"),
-   (0x1b, 0x1b, 0x1b, "image_atomic_inc"),
-   (0x1c, 0x1c, 0x1c, "image_atomic_dec"),
-   (0x1d, 0x1d,   -1, "image_atomic_fcmpswap"),
-   (0x1e, 0x1e,   -1, "image_atomic_fmin"),
-   (0x1f, 0x1f,   -1, "image_atomic_fmax"),
+   (0x0f, 0x0f, 0x10, 0x0f, "image_atomic_swap"),
+   (0x10, 0x10, 0x11, 0x10, "image_atomic_cmpswap"),
+   (0x11, 0x11, 0x12, 0x11, "image_atomic_add"),
+   (0x12, 0x12, 0x13, 0x12, "image_atomic_sub"),
+   (0x13,   -1,   -1,   -1, "image_atomic_rsub"),
+   (0x14, 0x14, 0x14, 0x14, "image_atomic_smin"),
+   (0x15, 0x15, 0x15, 0x15, "image_atomic_umin"),
+   (0x16, 0x16, 0x16, 0x16, "image_atomic_smax"),
+   (0x17, 0x17, 0x17, 0x17, "image_atomic_umax"),
+   (0x18, 0x18, 0x18, 0x18, "image_atomic_and"),
+   (0x19, 0x19, 0x19, 0x19, "image_atomic_or"),
+   (0x1a, 0x1a, 0x1a, 0x1a, "image_atomic_xor"),
+   (0x1b, 0x1b, 0x1b, 0x1b, "image_atomic_inc"),
+   (0x1c, 0x1c, 0x1c, 0x1c, "image_atomic_dec"),
+   (0x1d, 0x1d,   -1, 0x1d, "image_atomic_fcmpswap"),
+   (0x1e, 0x1e,   -1, 0x1e, "image_atomic_fmin"),
+   (0x1f, 0x1f,   -1, 0x1f, "image_atomic_fmax"),
 }
 # (gfx6, gfx7, gfx8, gfx9, gfx10, name) = (gfx6, gfx7, gfx89, gfx89, ???, name)
```

Comment:
looks like gfx7. So, I'd remove the gfx10 opcodes and just update the comment here and pick gfx7 below.

Reply:
Good point.